saleor/product/migrations/0183_calculate_discounted_prices.py

```diff
@@ -0,0 +1,402 @@
+    collections = fetch_collections(sale_model, pks)
+    channel_listings = fetch_sale_channel_listings(sale_listing_model, pks)
+    products = fetch_products(sale_model, pks)
+    categories = fetch_categories(sale_model, category_model, pks)
+    variants = fetch_variants(sale_model, pks)
+
+    return [
+        DiscountInfo(
+            sale=sale,
+            category_ids=categories[sale.pk],
+            channel_listings=channel_listings[sale.pk],
+            collection_ids=collections[sale.pk],
+            product_ids=products[sale.pk],
+            variants_ids=variants[sale.pk],
+        )
+        for sale in sales
+    ]
+
+
+def fetch_collections(sale_model, sale_pks):
+    collections = (
+        sale_model.collections.through.objects.filter(sale_id__in=sale_pks)
+        .order_by("id")
+        .values_list("sale_id", "collection_id")
+    )
+    collection_map = defaultdict(set)
+    for sale_pk, collection_pk in collections:
+        collection_map[sale_pk].add(collection_pk)
+    return collection_map
+
+
+def fetch_products(sale_model, sale_pks):
+    products = (
+        sale_model.products.through.objects.filter(sale_id__in=sale_pks)
+        .order_by("id")
+        .values_list("sale_id", "product_id")
+    )
+    product_map = defaultdict(set)
+    for sale_pk, product_pk in products:
+        product_map[sale_pk].add(product_pk)
+    return product_map
+
+
+def fetch_variants(
+    sale_model,
+    sale_pks,
+):
+    variants = (
+        sale_model.variants.through.objects.filter(sale_id__in=sale_pks)
+        .order_by("id")
+        .values_list("sale_id", "productvariant_id")
+    )
+    variants_map = defaultdict(set)
+    for sale_pk, variant_pk in variants:
+        variants_map[sale_pk].add(variant_pk)
+    return variants_map
+
+
+def fetch_sale_channel_listings(
+    sale_listing_model,
+    sale_pks,
+):
+    channel_listings = sale_listing_model.objects.filter(sale_id__in=sale_pks)
+    channel_listings_map = defaultdict(dict)
+    for channel_listing in channel_listings:
+        sale_id_row = channel_listings_map[channel_listing.sale_id]
+        sale_id_row[channel_listing.channel_id] = channel_listing
+    return channel_listings_map
+
+
+def fetch_categories(
+    sale_model,
+    category_model,
+    sale_pks,
+):
+    categories = (
+        sale_model.categories.through.objects.filter(sale_id__in=sale_pks)
+        .order_by("id")
+        .values_list("sale_id", "category_id")
+    )
+    category_map = defaultdict(set)
+    for sale_pk, category_pk in categories:
+        category_map[sale_pk].add(category_pk)
+    subcategory_map = defaultdict(set)
+    for sale_pk, category_pks in category_map.items():
+        subcategory_map[sale_pk] = set(
+            category_model.tree.filter(pk__in=category_pks)
+            .get_descendants(include_self=True)
+            .values_list("pk", flat=True)
+        )
+    return subcategory_map
+
+
+def update_discounted_prices_task(
```

Comment:
Should we remove `task` from the name of this function?

Reply:
Maybe we should store the original name in the comment if needed? But I don't know the use case when we could need to use the old name of this function.

Reply:
See `0041_add_calculation_transaction_events`, we are doing that in other migrations that have tasks in the previous version.